### PR TITLE
replace types-pkg_resources with types-setuptools

### PR DESCRIPTION
**Pull request by uri-granta**
**PR type:** bugfix

**Related issue(s)/PRs:** 

## Summary

replace types-pkg_resources was recently yanked from pypi, breaking the test pipeline

this PR replaces it with types-setuptools and fixes the revealed typing bugs

**Proposed changes**

**What alternatives have you considered?**

### Release notes

Pin to numpy<2.

**Fully backwards compatible:** yes

## PR checklist

- [ ] New features: code is well-documented
  - [ ] detailed docstrings (API documentation)
  - [ ] notebook examples (usage demonstration)
- [ ] The bug case / new feature is covered by unit tests
- [ ] Code has type annotations
- [ ] Build checks
  - [ ] I ran the black+isort formatter (`make format`)
  - [ ] I locally tested that the tests pass (`make check-all`)
- [ ] Release management
  - [ ] RELEASE.md updated with entry for this change
  - [ ] New contributors: I've added myself to CONTRIBUTORS.md

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
     "check_shapes>=1.0.0",
     "deprecated",
     "multipledispatch>=0.6",
-    "numpy",
+    "numpy<2",
```
Comment (avullo) [inline setup.py:14]: Out of curiosity, why is this necessary?

**Review by avullo — APPROVED** [review]
lgtm

File: setup.py (hunk 1 above)
Comment (uri-granta) [reply setup.py:14]: This should have already been there. We don't (yet) support numpy 2 — which came out a few months ago — in any of our toolboxes. There's an issue for adding support.
